framework/src/mfem/problem_operators/TimeDomainEquationSystemProblemOperator.C

```diff
@@ -28,34 +27,54 @@ TimeDomainEquationSystemProblemOperator::Init(mfem::BlockVector & X)
   GetEquationSystem()->BuildEquationSystem();
   // Set timestepper
   auto & ode_solver = _problem_data.ode_solver;
-  ode_solver = std::make_unique<mfem::BackwardEulerSolver>();
+  ode_solver = std::make_unique<MFEMBackwardEulerStateSolver>();
   ode_solver->Init(*(this));
   SetTime(_problem.time());
 }
 
 void
 TimeDomainEquationSystemProblemOperator::Solve()
 {
+  // Initialise time derivative
+  for (const auto i : index_range(_trial_var_names))
+  {
+    auto & trial_var_name = _trial_var_names.at(i);
+    auto & time_derivative_name =
+        _problem_data.time_derivative_map.getTimeDerivativeName(trial_var_name);
+    auto * trial_var = _problem_data.gridfunctions.Get(trial_var_name);
+    auto * trial_var_time_derivative = _problem_data.gridfunctions.Get(time_derivative_name);
+
+    *trial_var_time_derivative = *trial_var;
+  }
   // Advance time step of the MFEM problem. Time is also updated here, and
   // _problem_operator->SetTime is called inside the ode_solver->Step method to
   // update the time used by time dependent (function) coefficients.
   _problem_data.ode_solver->Step(_problem_data.f, _problem.time(), _problem.dt());
```

Comment:
Conceptually, I'd prefer us to do something like the following:
```suggestion
  _problem_data.ode_solver->Step(const_cast<mfem::Vector &>(*_trial_true_vector), _problem.time(), _problem.dt());
```
Or, otherwise, `Init()` should take no input.